src/Date.mo

```diff
@@ -140,32 +142,41 @@ module {
             t + (day - 1) * DAY + hour * HOUR + minute * MINUTE + second * SECOND + nano;
         };
 
-        private func toTextPadded(n : Nat) : Text {
-            if (n >= 10) {Nat.toText(n)} else { "0" # Nat.toText(n)}
+        private func padText(s : Text, desiredLength : Nat) : Text {
+            if (desiredLength <= s.size()) {return s};
+            padText("0" # s, desiredLength);
         };
 
-        public func isoFormat({
+        public func toIsoFormat({
             year; month; day; hour; minute; second; nano
         } : Date) : Text {
+            // Always outputs to millisecond precision
+            // Example "2016-02-29T23:59:59.000Z"
```

Comment:
What is the reason to always output millisecond precision even tho it is `0`?

Reply:
The rationale behind this was to enforce as much accuracy as possible from the original nanosecond timestamp. For example, even though 32 is numerically the same as 32.000, the latter implies that there are three decimal places that are accounted for, and that the number has not been rounded to the nearest second. Our `Date` type non-optionally accounts for nanoseconds, so it seemed that this would be the best choice to imply the existence of this extra information.

Also leads to less code complexity (adding an if/else).
Millisecond timestamps seem to be ubiquitous and widely supported. For example the LMS uses milliseconds by default.